opengrok-indexer/src/main/java/org/opengrok/indexer/analysis/AnalyzerGuru.java

```diff
@@ -1158,10 +1156,8 @@ private static String readOpening(InputStream in, byte[] sig)
 
             // If the opening starts with "#!", then track so that any
             // trailing whitespace after the hashbang is ignored.
-            if (opening.length() == 2) {
-                if (opening.charAt(0) == '#' && opening.charAt(1) == '!') {
-                    postHashbang = true;
-                }
+            if (opening.length() == 2 && opening.charAt(0) == '#' && opening.charAt(1) == '!') {
```

Comment:
Wouldn't `opening.equals("#!")` be nicer?

Reply:
Definitely nicer, thanks !

Reply:
My bad, I didn't notice that `opening` is a `StringBuilder`, in that case `toString` operation might be expensive if we execute it for every character.

Reply:
ok, reverting